src/coreComponents/fileIO/vtk/VTKPolyDataWriterInterface.hpp

```diff
@@ -251,6 +251,31 @@ class VTKPolyDataWriterInterface
   void writeElementFields( vtkSmartPointer< vtkCellData > celldata,
                            ElementRegionBase const & er ) const;
 
+  /*!
+   * @brief Writes all the fields with up to 3 dimensions associated to the elements of \p er if their plotlevel is <= m_plotLevel
+   * @param[in] celldata a VTK object containing all the fields associated with the elements
+   * @param[in] er ElementRegion being written
+   * @param[in] field the name of the field being written
+   */
+  template< class SUBREGION >
+  void writeElementField2D3D( vtkSmartPointer< vtkCellData > const celldata,
+                              ElementRegionBase const & er,
+                              string const & field ) const;
+
+  /*!
+   * @brief Writes all the 4D fields associated to the elements of \p er if their plotlevel is <= m_plotLevel
+   *        Considering that the second dimension has size N and the third dimension has size M, this
+   *        function will write MxN fields to VTK
+   * @param[in] celldata a VTK object containing all the fields associated with the elements
```

Comment:
Can you elaborate more on what you are currently doing? On what will the output be?

Reply:
I added more comments on what people will be able to click on in Paraview. The Paraview fields will look like:
```
            field_0_0
              |
            field_0_(M-1)
              |
            field_(N-1)_0
              |
            field_(N-1)_(M-1)
```
and if the dimension of the 2nd dimension is one (which is the case for fluid constitutive data for flow-only simulations), then the first _0 is skipped. For a CO2-brine simulation, we will be able to click for instance:
```    
          phaseCompFraction_0
          phaseCompFraction_1
```
so this just mimics what is done in Silo/Visit.